### PR TITLE
Add pushsafer.com notification service

### PR DESCRIPTION
**Pull request by appzer**
Pushsafer
# irssi-pushsafer

Plugin for irssi (a console based IRC client) to send push-notifications using https://www.pushsafer.com.

This allows you to be notified when someone messages/mentions you on IRC, 
when you're not online.

# Installation

  1. register or login inot your pushsafer control panel abd note your private or alias key.
  2. cp pushsafer.pl to ~/.irssi/scripts/ and symlink into scripts/autorun if you desire.
  3. touch ~/.irssi/pushsafer_ignores
  4. Within irssi: 
    1. /load autorun/pushsafer.pl
    2. /set pushsafer_key "your pushsafer private or alias key"
    3. optional set the following params https://www.pushsafer.com/en/pushapi
    4. /set pushsafer_device "device or device group id"
    5. /set pushsafer_sound "sound number"
    6. /set pushsafer_icon "icon number>
    7. /set pushsafer_vibration "vibration 0-3>
    8. /set pushsafer_url "optional url"
    9. /set pushsafer_u

### CODE REVIEW

File: scripts/pushsafer.pl [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+use Irssi;
+use vars qw($VERSION %IRSSI %config);
+use LWP::UserAgent;
+use Scalar::Util qw(looks_like_number);
+
+$VERSION = '0.0.1';
+
+%IRSSI = (
+    authors => 'Pushsafer.com Kevin Siml',
```
Comment (ailin-nemui) [inline scripts/pushsafer.pl]: you do not need to include your website in the authors field

File: scripts/pushsafer.pl [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+   my ($data, $server, $item) = @_;
+   $data =~ s/^([\s]+).*$/$1/;
+   my $orig_debug = Irssi::settings_get_bool('pushsafer_debug');
+   Irssi::settings_set_bool('pushsafer_debug', 1);
+   debug("Sending test message :" . $data);
+   send_push("Test Message", strip_formating($data));
+   Irssi::settings_set_bool('pushsafer_debug', $orig_debug);
+}
+
+sub strip_formating {
```
Comment (ailin-nemui) [inline scripts/pushsafer.pl]: please test if Irssi::strip_codes works as a replacement for your custom function?

File: scripts/pushsafer.pl [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        return;
+    }
+    print $fp join("\n", @_);
+    close $fp;
+}
+
+sub read_file {
+    read_settings();
+    my $fp;
+    if (!open($fp, "<", $pushsafer_ignorefile)) {
```
Comment (ailin-nemui) [inline scripts/pushsafer.pl]: maybe you should just check if an ignore file exists and return silently if that is not a case, that would safe the user having to "touch" an empty file

**Comment by appzer** [comment]
@ailin-nemui 
Had made all the proposed changes

**Pull request merged**